src/syntax_util.ml

```diff
@@ -137,34 +137,68 @@ let syntax_error_extension_node loc message =
  in
  (str, payload)
 
+let string_prefix ~prefix s =
```

Comment:
This could use a comment explaining what `string_prefix` does.

Reply:
And could it ever be the case that `s` is smaller in length than `prefix` (causing runtime error)? For example, an identifier such as `"matc"`. Do you want to do something like:

```
let matched = ref true in
   let n = String.length prefix in
   if String.length s >= n then
     for i = 0 to n - 1 do
       matched := !matched && prefix.[i] = s.[i]
     done;
     !matched
  else
    false
```

Reply:
Yeah - sorry about that! That's the bug I left a comment about. I realized I hadn't handled that case just after I turned off my computer.

This could also short-circuit by raising and immediately catching an exception when a character doesn't match.

I'll fix these issues and add some comments when I'm in front of a computer again, hopefully tomorrow.

Reply:
Thanks for taking the time to review!